llvm/test/CodeGen/AMDGPU/atomic_optimizations_pixelshader.ll

```diff
@@ -16,7 +16,8 @@ declare void @llvm.amdgcn.raw.ptr.buffer.store.f32(float, ptr addrspace(8), i32,
 define amdgpu_ps void @add_i32_constant(ptr addrspace(8) inreg %out, ptr addrspace(8) inreg %inout) {
 ; GFX7-LABEL: add_i32_constant:
 ; GFX7:       ; %bb.0: ; %entry
-; GFX7-NEXT:    s_mov_b64 s[10:11], exec
+; GFX7-NEXT:    s_mov_b64 s[8:9], exec
```

Comment:
Why does this patch cause any codegen changes?

Reply:
Conditions for introducing a variable live mask were slightly different.
I have match them pre-refactor and made them more readable.